aioshelly/rpc_device.py

```diff
@@ -2,18 +2,21 @@
 from __future__ import annotations
 
 import asyncio
-from typing import Any, Dict
+from typing import Any, Callable
 
 import aiohttp
+from aiohttp.client import ClientSession
 
 from .common import ConnectionOptions, IpOrOptionsType, get_info, process_ip_or_options
-from .exceptions import AuthRequired, NotInitialized, WrongShellyGen
+from .exceptions import AuthRequired, NotInitialized, ShellyError, WrongShellyGen
 from .wsrpc import WsRPC
 
 
-def mergedicts(dict1, dict2):
+def mergedicts(dict1: dict, dict2: dict | None) -> dict:
```

Comment:
No need to allow `None` here for `dict2`
```suggestion
def mergedicts(dict1: dict, dict2: dict) -> dict:
```